src/Host/Client/Impl/Host/BrokerClient.cs

```diff
@@ -133,8 +139,8 @@ internal abstract class BrokerClient : IBrokerClient {
                 } catch (UnauthorizedAccessException) {
                     isValidCredentials = false;
                     continue;
-                } catch (HttpRequestException ex) {
-                    throw new RHostDisconnectedException("HTTP error while creating session: " + ex.Message, ex);
+                } catch (BrokerApiErrorException apiex) {
+                    throw new RHostDisconnectedException(apiex.ApiError);
```

Comment:
I am afraid to break things here. There are few places than expect RHostDisconnectedException to contain the proper message.